Functions.Templates/Templates-v2/EventHubTrigger-Python/template.json

```diff
@@ -0,0 +1,228 @@
+            ]
+        },
+        {
+            "name": "Add EventHubTrigger function to the main file",
+            "actions": [
+                "getFunctionName",
+                "getEventHubName",
+                "getConnnectionString",
+                "readFileContent_FunctionBody",
+                "replaceTokensInText_FunctionBody",
+                "appendFileContent_FunctionApp"
+            ]
+        },
+        {
+            "name": "Create New Blueprint file",
+            "input" : {
+                "userCommand" : "createBluePrint",
+                "assignTo" : "$(BLUEPRINT_FILENAME)"
+            },
+            "actions": [
+                "getBluePrintFileName",
+                "getFunctionName",
+                "getEventHubName",
+                "getConnnectionString",
+                "readFileContent_BlueprintFile",
+                "replaceTokensInText_BlueprintFile",
+                "readFileContent_BlueprintBody",
+                "replaceTokensInText_BlueprintBody",
+                "writeFile_BlueprintFile",
+                "appendFileContent_BlueprintBody",
+                "readFileContent_BlueprintImport",
+                "replaceTokensInText_BlueprintImport",
+                "writeFile_BlueprintImport",
+                "appendFileContent_FunctionApp",
+                "readFileContent_BlueprintRegister",
+                "replaceTokensInText_BlueprintRegister"
+            ]
+        },
+        {
+            "name": "Add EventHubTrigger function to the Blueprint",
+            "actions": [
+                "getBluePrintFileName",
+                "getFunctionName",
+                "getEventHubName",
+                "getConnnectionString",
+                "readFileContent_BlueprintBody",
+                "replaceTokensInText_BlueprintBody",
+                "appendFileContent_BlueprintBody"
+            ]
+        }
+    ],
+    "actions" : [
+        {
+            "name": "getFunctionName",
+            "type": "UserInput",
+            "assignTo": "$(FUNCTION_NAME_INPUT)",
+            "paramId": "trigger-functionName",
+            "defaultValue": "EventhubTrigger"
+        },
+        {
+            "name": "getEventHubName",
+            "type": "UserInput",
+            "assignTo": "$(EVENTHUB_NAME)",
+            "paramId": "trigger-eventhubName",
+            "defaultValue": "MyEventHub"
+        },
+        {
+            "name": "getConnnectionString",
+            "type": "UserInput",
+            "assignTo": "$(EVENTHUB_CONNECTION_STRING)",
+            "paramId": "trigger-connectionStringName",
+            "defaultValue": ""
+        },
+        {
+            "name": "readFileContent_FunctionApp",
+            "type": "ReadFromFile",
+            "assignTo": "$(FUNCTION_APP_CONTENT)",
+            "filePath" : "function_app.py"
+        },
+        {
+            "name": "replaceTokensInText_FunctionApp",
+            "type": "ReplaceTokensInText",

```

Comment:
A few things about the `ReplaceTokensInText` action:
1. I'm worried that it will be easy to mess up the ordering on this (aka you accidentally do it before a user prompt or after writing to a file)
3. It feels very boilerplate-y. The source and assignTo are always the same, there's no unique properties, etc.

What if `ReplaceTokensInText` was an implied part of `WriteToFile` and `AppendToFile` instead of an explicit action? That way you'll never forget it, you'll always do it at the right time, and we can reduce some boilerplate.

Reply:
We have updated `WriteToFile` to include addtional property called `replaceTokens` that is bool and is true by default. We discussed some scenarios where having `ReplaceTokensInText` as an explicit action is still valuable therefore decided to keep that action as well.